includes/Modules/Reader_Revenue_Manager.php

```diff
@@ -225,6 +231,73 @@ protected function create_data_request( Data_Request $data ) {
 				 */
 				$subscribewithgoogle = $this->get_service( 'subscribewithgoogle' );
 				return $subscribewithgoogle->publications->listPublications( array( 'filter' => $this->get_publication_filter() ) );
+
+			case 'POST:sync-publication-onboarding-state':
+				if ( empty( $data['publicationID'] ) ) {
+					throw new Missing_Required_Param_Exception( 'publicationID' );
+				}
+
+				if ( empty( $data['publicationOnboardingState'] ) ) {
+					throw new Missing_Required_Param_Exception( 'publicationOnboardingState' );
+				}
+
+				$return_data  = array();
+				$publications = $this->get_data( 'publications' );
+
+				if ( is_wp_error( $publications ) ) {
+					return $publications;
+				}
+
+				if ( empty( $publications ) ) {
+					return new WP_Error(
+						'publication_not_found',
+						__( 'Publication not found.', 'google-site-kit' ),
+						array( 'status' => 404 )
+					);
+				}
+
+				$publication = array_filter(
+					$publications,
+					function ( $publication ) use ( $data ) {
+						return $publication->getPublicationId() === $data['publicationID'];
+					}
+				);
+
+				if ( empty( $publication ) ) {
+					return new WP_Error(
+						'publication_not_found',
+						__( 'Publication not found.', 'google-site-kit' ),
+						array( 'status' => 404 )
+					);
+				}
+
+				$publication          = reset( $publication );
+				$new_onboarding_state = $publication->getOnboardingState();
+
+				if ( $new_onboarding_state === $data['publicationOnboardingState'] ) {
+					return function () use ( $return_data ) {
+						return (object) $return_data;
+					};
```

Comment:
As per my comment above we don't need to refer to `$return_data` here, it's easier to follow if we just return an empty array directly:

```suggestion
					return function ()  {
						return (object) array();
					};
```